src/NuGet.Core/NuGet.Packaging/Signing/Utility/X509ChainHolder.cs

```diff
@@ -20,13 +20,31 @@ public sealed class X509ChainHolder : IDisposable
 
         public X509Chain Chain { get; }
 
+        [Obsolete("Use either CreateForCodeSigning() or CreateForTimestamping() instead.")]
         public X509ChainHolder()
         {
-            IX509ChainFactory creator = X509TrustStore.GetX509ChainFactory(NullLogger.Instance);
+            IX509ChainFactory creator = X509TrustStore.GetX509ChainFactory(X509StorePurpose.CodeSigning, NullLogger.Instance);
 
             Chain = creator.Create();
         }
 
+        private X509ChainHolder(X509StorePurpose storePurpose)
+        {
+            IX509ChainFactory creator = X509TrustStore.GetX509ChainFactory(storePurpose, NullLogger.Instance);
+
+            Chain = creator.Create();
+        }
+
+        public static X509ChainHolder CreateForCodeSigning()
```

Comment:
Is it work making this `internal` and updating `InternalsVisibleTo` so that the unit tests can access it?

Reply:
+1.  A while ago I wrote https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md#changes-to-apis. If the APIs are not useful to customers using NuGet.Packaging in their own apps, then it probably shouldn't be a public API.

Reply:
`X509ChainHolder` is already public.  This PR changes the meaning of the default constructor (a CS-only chain instead of a CS+TS chain) without access to the new behavior (a TS-only chain).  Hence the new `public` methods.

I'll remove the `Obsolete` attribute and make these 2 methods `internal`.  We revisit this later based on feedback.